### ISSUE TITLE
`set-column!` fail on `:vectorz` matrix.

### ISSUE THREAD

**Issue by harold** [mikera/core.matrix#308]
With:
```
[net.mikera/core.matrix "0.57.0"]
[net.mikera/vectorz-clj "0.45.0"]
```

I can set the second, but not the third, column of this matrix:
```
> (m/set-column! (m/matrix :vectorz [[1 2 3] [4 5 6]]) 1 [0 0])
#<mikera.matrixx.Matrix@3d477ece [[1.0,0.0,3.0],
                                  [4.0,0.0,6.0]]>

> (m/set-column! (m/matrix :vectorz [[1 2 3] [4 5 6]]) 2 [0 0])
IndexOutOfBoundsException Matrix with shape [2,3] does not have slice: 2 on dimension 0  mikera.matrixx.impl.ARectangularMatrix.checkRow (ARectangularMatrix.java:93)
```

**Comment by harold** [comment]
@mikera --- hope everything's cool in your world.

I'd love to know your thoughts on this, I poked around a bit and it seems that `.checkLength` on `AVector` is overridden by `.checkRow` in this case (or something?); checking rows while setting columns seems a bit weird (tbh, I don't really know how to follow the Java code, so I could be way off).

**Comment by harold** [comment]
@charlesg3 --- relevant to your interests.

**Comment by schaueho** [comment]
I reported the exact same issue: https://github.com/mikera/vectorz-clj/issues/63 and @mikera fixed it in vectorz-clj 0.46.0 (and vectorz 0.65.0 if I read the changeset correctly).

**Comment by harold** [comment]
@schaueho --- funny coincidence! Good looking out.
